docs/proposals/coap-core.md

```diff
@@ -0,0 +1,74 @@
+
+## Discovery Process
+
+### The standard
+
+A well-known relative URI `/.well-known/core` is defined as a default entry point for requesting the list of links about resources hosted by a server and thus performing CoRE Resource Discovery. The CoRE protocol is applicable for use with Constrained Application Protocol (CoAP) [COAP](https://tools.ietf.org/html/rfc7252), HTTP, or any other suitable web transfer protocol.
+
+Resource Discovery can be performed either unicast or multicast.
+
+When a server’s IP address is already known, either a priori or resolved via the Domain Name System (DNS), unicast discovery is performed in order to locate the entry point to the resource of interest. In this specification, this is performed using a GET to `/.well-known/core` on the server, which returns a payload in the CoRE Link Format.
+
+Multicast Resource Discovery is useful when a client needs to locate a resource within a limited scope, and that scope supports IP multicast. A GET request to the appropriate multicast address is made for `/.well-known/core`. In order to limit the number and size of responses, a query string is recommended with the known attributes. Typically, a resource would be discovered based on its Resource Type and/or Interface Description, along with possible application-specific attributes.
+
+### With Akri
+
+The Akri Configuration defines a list of IP addresses to use for resource discovery, implementing thus unicast discovery. Adding support to multicast discovery is surely desiderable.
+
+1. Akri agent sends a `GET /well-known/core` request to the device at `coap://{IP_ADDRESS}:5683`. The standard defines that CoAP devices which mean to support resource discovery, must be reachable with the default port `:5683` and expose endpoint `/well-known/core` implementing the CoRE Link Format.
+2. The device responds with the list of supported resources. An example of response in link format is the following:
+    ```
+    </sensors/temp>;rt="oic.r.temperature";if="sensor",
+    </sensors/light>;rt="oic.r.light.brightness";if="sensor"
+    ```
+
+    The example is stating that the device has 2 REST resources, `/sensors/temp` and `/sensors/light` which are of type `oic.r.temperature` and `oic.r.light.brightness` respectively. `rt` values are defined in [IANA](https://www.iana.org/assignments/core-parameters/core-parameters.xhtml#rt-link-target-att-value) to have same standardization, although vendor values can probably be used. Then `if` means that the interface description of the resource is of type `sensor`. Currently only resources with interface `sensor` are supported by the agent.
```

Comment:
Do you see Akri supporting [other interfaces](https://www.iana.org/assignments/core-parameters/core-parameters.xhtml#if-link-target-att-value), such as the actuator one?

Reply:
I'd personally leave this for someone else for the time being. Maybe someone else will have an use case for it in future and will contribute :)

Reply:
Good point. This could definitely come in a later iteration.